cmd/nerdctl/builder_build_linux_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/containerd/nerdctl/v2/pkg/testutil"
+	"gotest.tools/v3/assert"
+)
+
+func TestBuildContextWithOCILayout(t *testing.T) {
+	testutil.RequiresBuild(t)
+	testutil.RegisterBuildCacheCleanup(t)
+
+	var dockerBuilderArgs []string
+	if testutil.IsDocker() {
+		// Default docker driver does not support OCI exporter.
+		// Reference: https://docs.docker.com/build/exporters/oci-docker/
+		builderName := testutil.SetupDockerContainerBuilder(t)
+		dockerBuilderArgs = []string{"buildx", "--builder", builderName}
+	}
+
+	base := testutil.NewBase(t)
+	imageName := testutil.Identifier(t)
+	t.Cleanup(func() { base.Cmd("rmi", imageName).AssertOK() })
```

Comment:
Generally speaking, I would suggest you **also** run your cleanup before starting the test.

The reason is that `t.Cleanup` is not always called, and you may end-up in with a broken environment that fails test.

Pretty much

```
tearDown := func() { base.Cmd("rmi", imageName).AssertOK() }

t.Cleanup(tearDown)
tearDown()
```

In the case here

Reply:
Also, it is fine to fail in the cleanup procedure. So, `Run` instead of `AssertOK` will do here.

Reply:
Sure, sounds good. Should be addressed in latest push.

Reply:
Feel free to hit "resolve" on these comments to hide them (just to keep the progress/conversation more clear).